tests/test_bookmark.py

```diff
@@ -105,3 +106,19 @@ def _PartBookmarkFinder_(self, request):
     @pytest.fixture
     def document_part_(self, request):
         return instance_mock(request, DocumentPart)
+
+
+class Describe_PartBookmarkFinder(object):
+    def it_calls_iter_start_end_pairs_of_a_story(self, document_part_,
+                                                 _iter_start_end_pairs):
+        _PartBookmarkFinder.iter_start_end_pairs(document_part_)
+        _iter_start_end_pairs.assert_called_once_with()
```

Comment:
This situation is a very common pattern, occurring whenever a Function Object like this is implemented. The basic unit test pattern goes like this:

```python
def it_provides_an_iter_start_end_pairs_interface_method(
        self, part_, _init_, _iter_start_end_pairs_)

    pairs = _PartBookmarkFinder.iter_start_end_pairs(part_)

    # ---ANY is required here because we cannot know the `self` object from here---
    _init_.assert_called_once_with(ANY, part_)
    _iter_start_end_pairs_.assert_called_once_with(ANY)
    assert pairs == _iter_start_end_pairs_.return_value
```

You want to use a mock `Part` instead of `DocumentPart` because the method is not specific, it's intended to operate on any part.